presto-main/src/main/java/com/facebook/presto/cost/ConnectorFilterStatsCalculatorService.java

```diff
@@ -111,7 +111,8 @@ private static ColumnStatistics toColumnStatistics(VariableStatsEstimate variabl
             builder.setDistinctValuesCount(Estimate.of(variableStatsEstimate.getDistinctValuesCount()));
         }
 
-        if (!Double.isNaN(variableStatsEstimate.getAverageRowSize())) {
+        // for (synthesized) bucket columns, averageRowSize is not null (4, derived from columnType int) and nullFraction is NULL.
+        if (!Double.isNaN(variableStatsEstimate.getAverageRowSize()) && !Double.isNaN(nonNullRowsCount)) {
```

Comment:
In what case the stats are missing? Why didn't we need this special case before?

Reply:
Before, it is assumed (not strictly guaranteed) that averageRowSize and nonNullRowCount are either all NaN, or all have meaningful values. And it is not solid anyway.

For our case, as we derive the averageRowSize from the columnType already, it becomes much more likely that averageRowSize is not NaN, but nullRowCount is still NaN. And this is why we need to add this additional condition.

Reply:
Could you please elaborate a little bit more on what was the issue and it what cases it manifests?

Reply:
Sorry, i commented on the wrong line